entity-data-service-rx-client/src/main/java/org/hypertrace/entity/data/service/rxclient/EntityDataCachingClient.java

```diff
@@ -112,9 +121,14 @@
     private UpsertCondition condition;
 
     private void executeUpdate() {
-
-      EntityDataCachingClient.this.pendingEntityUpdates.remove(entityKey);
-
+      // Acquire write lock to ensure no more modification of this update
+      Lock lock = pendingUpdateStripedLock.get(entityKey).writeLock();
+      try {
+        lock.lock();
+        EntityDataCachingClient.this.pendingEntityUpdates.remove(entityKey);
+      } finally {
+        lock.unlock();
```

Comment:
Is it ok to unlock immediately after removal or should we do the rest of the processing?

Reply:
It should be - I initially had it on the whole method, but since the write lock is exclusive it seems like it'd be better to scope it. As soon as the pending update is removed from the map, future reads should be good to go as they'll trigger the creation of a new update.

Reply:
ok.